[native] hibernate-search-orm-elasticsearch extension not working with GraalVM 20.3-dev

## Describe the bug
As of 7f8617e6767bc435f19aae198614b617e9703a65 native apps using `hibernate-search-orm-elasticsearch` cannot be compiled using GraalVM/Mandrel dev versions.

The issue is caused by the usage of `graalVMVersion.isSnapshot()` as a mean to see if the JVM is GraalVM or not.
`graalVMVersion.isSnapshot()` however is not meant to be used for that purpose and returns true not only in OpenJDK but in GraalVM dev versions as well.

### Expected behavior
Native apps should compile and run.

### Actual behavior

<details>
<summary>The native image generation fails with:</summary>
<pre>
Error: Unsupported features in 4 methods
Detailed message:
Error: com.oracle.graal.pointsto.constraints.UnsupportedFeatureException: Invoke with MethodHandle argument could not be reduced to at most a single call or single field access. The method handle must be a compile time constant, e.g., be loaded from a `static final` field. Method that contains the method handle invocation: java.lang.invoke.MethodHandle.invokeBasic(Object)
To enable method handles that do not require LambdaForm interpretation (e.g. because of a call to MethodHandle.bindTo()) or to diagnose the issue, you can add the option --report-unsupported-elements-at-runtime. The error is then reported at run time when the invoke is executed and the method handle has to be interpreted.
Trace: 
	at parsing java.lang.invoke.LambdaForm$MH/0x00000007c24ac040.invoke_MT(LambdaForm$MH)
Call path from entry point to java.lang.invoke.LambdaForm$MH/1574591223.invoke_MT(Object, Object, Object): 
	at java.lang.invoke.LambdaForm$MH/0x00000007c24ac040.invoke_MT(LambdaForm$MH)
	at org.hibernate.search.util.common.reflect.impl.MethodHandleValueReadHandle.get(MethodHandleValueReadHandle.java:39)
	at org.hibernate.search.mapper.pojo.bridge.runtime.impl.PropertyIdentifierMapping.getIdentifier(PropertyIdentifierMapping.java:60)
	at org.hibernate.search.mapper.pojo.work.impl.PojoTypeIndexer.add(PojoTypeIndexer.java:37)
	at org.hibernate.search.mapper.pojo.work.impl.PojoIndexerImpl.add(PojoIndexerImpl.java:42)
	at org.hibernate.search.mapper.orm.massindexing.impl.IdentifierConsumerDocumentProducer.index(IdentifierConsumerDocumentProducer.java:243)
	at org.hibernate.search.mapper.orm.massindexing.impl.IdentifierConsumerDocumentProducer.indexList(IdentifierConsumerDocumentProducer.java:201)
	at org.hibernate.search.mapper.orm.massindexing.impl.IdentifierConsumerDocumentProducer.loadAndIndexList(IdentifierConsumerDocumentProducer.java:154)
	at org.hibernate.search.mapper.orm.massindexing.impl.IdentifierConsumerDocumentProducer.loadAndIndexAllFromQueue(IdentifierConsumerDocumentProducer.java:123)
	at org.hibernate.search.mapper.orm.massindexing.impl.IdentifierConsumerDocumentProducer.run(IdentifierConsumerDocumentProducer.java:103)
	at java.lang.Thread.run(Thread.java:829)
	at com.oracle.svm.core.thread.JavaThreads.threadStartRoutine(JavaThreads.java:519)
	at com.oracle.svm.core.posix.thread.PosixJavaThreads.pthreadStartRoutine(PosixJavaThreads.java:192)
	at com.oracle.svm.core.code.IsolateEnterStub.PosixJavaThreads_pthreadStartRoutine_e1f4a8c0039f8337338252cd8734f63a79b5e3df(generated:0)
Error: com.oracle.svm.hosted.substitute.DeletedElementException: Unsupported type java.lang.invoke.MemberName is reachable: All methods from java.lang.invoke should have been replaced during image building.
To diagnose the issue, you can add the option --report-unsupported-elements-at-runtime. The unsupported element is then reported at run time when it is accessed the first time.
Trace: 
	at parsing java.lang.invoke.LambdaForm$NamedFunction.equals(LambdaForm.java:1177)
Call path from entry point to java.lang.invoke.LambdaForm$NamedFunction.equals(Object): 
	at java.lang.invoke.LambdaForm$NamedFunction.equals(LambdaForm.java:1173)
	at java.util.HashMap.getNode(HashMap.java:574)
	at java.util.HashMap.get(HashMap.java:552)
	at com.oracle.svm.core.jdk.Target_java_nio_charset_Charset.lookup(CharsetSubstitutions.java:74)
	at java.nio.charset.Charset.isSupported(Charset.java:503)
	at com.oracle.svm.jni.JNIJavaCallWrappers.jniInvoke_VA_LIST:Ljava_nio_charset_Charset_2_0002eisSupported_00028Ljava_lang_String_2_00029Z(generated:0)
Error: com.oracle.svm.hosted.substitute.DeletedElementException: Unsupported type java.lang.invoke.MemberName is reachable: All methods from java.lang.invoke should have been replaced during image building.
To diagnose the issue, you can add the option --report-unsupported-elements-at-runtime. The unsupported element is then reported at run time when it is accessed the first time.
Trace: 
	at parsing java.lang.invoke.LambdaForm$NamedFunction.hashCode(LambdaForm.java:1182)
Call path from entry point to java.lang.invoke.LambdaForm$NamedFunction.hashCode(): 
	at java.lang.invoke.LambdaForm$NamedFunction.hashCode(LambdaForm.java:1182)
	at java.util.HashMap.hash(HashMap.java:339)
	at java.util.HashMap.get(HashMap.java:552)
	at com.oracle.svm.core.jdk.Target_java_nio_charset_Charset.lookup(CharsetSubstitutions.java:74)
	at java.nio.charset.Charset.isSupported(Charset.java:503)
	at com.oracle.svm.jni.JNIJavaCallWrappers.jniInvoke_VA_LIST:Ljava_nio_charset_Charset_2_0002eisSupported_00028Ljava_lang_String_2_00029Z(generated:0)
Error: com.oracle.svm.hosted.substitute.DeletedElementException: Unsupported type java.lang.invoke.MemberName is reachable: All methods from java.lang.invoke should have been replaced during image building.
To diagnose the issue, you can add the option --report-unsupported-elements-at-runtime. The unsupported element is then reported at run time when it is accessed the first time.
Trace: 
	at parsing java.lang.invoke.LambdaForm$NamedFunction.toString(LambdaForm.java:1279)
Call path from entry point to java.lang.invoke.LambdaForm$NamedFunction.toString(): 
	at java.lang.invoke.LambdaForm$NamedFunction.toString(LambdaForm.java:1279)
	at java.lang.String.valueOf(String.java:2951)
	at java.io.PrintStream.print(PrintStream.java:745)
	at java.io.PrintStream.println(PrintStream.java:882)
	at com.oracle.svm.jni.functions.JNIFunctions.ExceptionDescribe(JNIFunctions.java:757)
	at com.oracle.svm.core.code.IsolateEnterStub.JNIFunctions_ExceptionDescribe_b5412f7570bccae90b000bc37855f00408b2ad73(generated:0)

com.oracle.svm.core.util.UserError$UserException: Unsupported features in 4 methods
Detailed message:
Error: com.oracle.graal.pointsto.constraints.UnsupportedFeatureException: Invoke with MethodHandle argument could not be reduced to at most a single call or single field access. The method handle must be a compile time constant, e.g., be loaded from a `static final` field. Method that contains the method handle invocation: java.lang.invoke.MethodHandle.invokeBasic(Object)
To enable method handles that do not require LambdaForm interpretation (e.g. because of a call to MethodHandle.bindTo()) or to diagnose the issue, you can add the option --report-unsupported-elements-at-runtime. The error is then reported at run time when the invoke is executed and the method handle has to be interpreted.
Trace: 
	at parsing java.lang.invoke.LambdaForm$MH/0x00000007c24ac040.invoke_MT(LambdaForm$MH)
Call path from entry point to java.lang.invoke.LambdaForm$MH/1574591223.invoke_MT(Object, Object, Object): 
	at java.lang.invoke.LambdaForm$MH/0x00000007c24ac040.invoke_MT(LambdaForm$MH)
	at org.hibernate.search.util.common.reflect.impl.MethodHandleValueReadHandle.get(MethodHandleValueReadHandle.java:39)
	at org.hibernate.search.mapper.pojo.bridge.runtime.impl.PropertyIdentifierMapping.getIdentifier(PropertyIdentifierMapping.java:60)
	at org.hibernate.search.mapper.pojo.work.impl.PojoTypeIndexer.add(PojoTypeIndexer.java:37)
	at org.hibernate.search.mapper.pojo.work.impl.PojoIndexerImpl.add(PojoIndexerImpl.java:42)
	at org.hibernate.search.mapper.orm.massindexing.impl.IdentifierConsumerDocumentProducer.index(IdentifierConsumerDocumentProducer.java:243)
	at org.hibernate.search.mapper.orm.massindexing.impl.IdentifierConsumerDocumentProducer.indexList(IdentifierConsumerDocumentProducer.java:201)
	at org.hibernate.search.mapper.orm.massindexing.impl.IdentifierConsumerDocumentProducer.loadAndIndexList(IdentifierConsumerDocumentProducer.java:154)
	at org.hibernate.search.mapper.orm.massindexing.impl.IdentifierConsumerDocumentProducer.loadAndIndexAllFromQueue(IdentifierConsumerDocumentProducer.java:123)
	at org.hibernate.search.mapper.orm.massindexing.impl.IdentifierConsumerDocumentProducer.run(IdentifierConsumerDocumentProducer.java:103)
	at java.lang.Thread.run(Thread.java:829)
	at com.oracle.svm.core.thread.JavaThreads.threadStartRoutine(JavaThreads.java:519)
	at com.oracle.svm.core.posix.thread.PosixJavaThreads.pthreadStartRoutine(PosixJavaThreads.java:192)
	at com.oracle.svm.core.code.IsolateEnterStub.PosixJavaThreads_pthreadStartRoutine_e1f4a8c0039f8337338252cd8734f63a79b5e3df(generated:0)
Error: com.oracle.svm.hosted.substitute.DeletedElementException: Unsupported type java.lang.invoke.MemberName is reachable: All methods from java.lang.invoke should have been replaced during image building.
To diagnose the issue, you can add the option --report-unsupported-elements-at-runtime. The unsupported element is then reported at run time when it is accessed the first time.
Trace: 
	at parsing java.lang.invoke.LambdaForm$NamedFunction.equals(LambdaForm.java:1177)
Call path from entry point to java.lang.invoke.LambdaForm$NamedFunction.equals(Object): 
	at java.lang.invoke.LambdaForm$NamedFunction.equals(LambdaForm.java:1173)
	at java.util.HashMap.getNode(HashMap.java:574)
	at java.util.HashMap.get(HashMap.java:552)
	at com.oracle.svm.core.jdk.Target_java_nio_charset_Charset.lookup(CharsetSubstitutions.java:74)
	at java.nio.charset.Charset.isSupported(Charset.java:503)
	at com.oracle.svm.jni.JNIJavaCallWrappers.jniInvoke_VA_LIST:Ljava_nio_charset_Charset_2_0002eisSupported_00028Ljava_lang_String_2_00029Z(generated:0)
Error: com.oracle.svm.hosted.substitute.DeletedElementException: Unsupported type java.lang.invoke.MemberName is reachable: All methods from java.lang.invoke should have been replaced during image building.
To diagnose the issue, you can add the option --report-unsupported-elements-at-runtime. The unsupported element is then reported at run time when it is accessed the first time.
Trace: 
	at parsing java.lang.invoke.LambdaForm$NamedFunction.hashCode(LambdaForm.java:1182)
Call path from entry point to java.lang.invoke.LambdaForm$NamedFunction.hashCode(): 
	at java.lang.invoke.LambdaForm$NamedFunction.hashCode(LambdaForm.java:1182)
	at java.util.HashMap.hash(HashMap.java:339)
	at java.util.HashMap.get(HashMap.java:552)
	at com.oracle.svm.core.jdk.Target_java_nio_charset_Charset.lookup(CharsetSubstitutions.java:74)
	at java.nio.charset.Charset.isSupported(Charset.java:503)
	at com.oracle.svm.jni.JNIJavaCallWrappers.jniInvoke_VA_LIST:Ljava_nio_charset_Charset_2_0002eisSupported_00028Ljava_lang_String_2_00029Z(generated:0)
Error: com.oracle.svm.hosted.substitute.DeletedElementException: Unsupported type java.lang.invoke.MemberName is reachable: All methods from java.lang.invoke should have been replaced during image building.
To diagnose the issue, you can add the option --report-unsupported-elements-at-runtime. The unsupported element is then reported at run time when it is accessed the first time.
Trace: 
	at parsing java.lang.invoke.LambdaForm$NamedFunction.toString(LambdaForm.java:1279)
Call path from entry point to java.lang.invoke.LambdaForm$NamedFunction.toString(): 
	at java.lang.invoke.LambdaForm$NamedFunction.toString(LambdaForm.java:1279)
	at java.lang.String.valueOf(String.java:2951)
	at java.io.PrintStream.print(PrintStream.java:745)
	at java.io.PrintStream.println(PrintStream.java:882)
	at com.oracle.svm.jni.functions.JNIFunctions.ExceptionDescribe(JNIFunctions.java:757)
	at com.oracle.svm.core.code.IsolateEnterStub.JNIFunctions_ExceptionDescribe_b5412f7570bccae90b000bc37855f00408b2ad73(generated:0)

	at com.oracle.svm.core.util.UserError.abort(UserError.java:82)
	at com.oracle.svm.hosted.FallbackFeature.reportAsFallback(FallbackFeature.java:217)
	at com.oracle.svm.hosted.NativeImageGenerator.runPointsToAnalysis(NativeImageGenerator.java:766)
	at com.oracle.svm.hosted.NativeImageGenerator.doRun(NativeImageGenerator.java:556)
	at com.oracle.svm.hosted.NativeImageGenerator.lambda$run$0(NativeImageGenerator.java:469)
	at java.base/java.util.concurrent.ForkJoinTask$AdaptedRunnableAction.exec(ForkJoinTask.java:1407)
	at java.base/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:290)
	at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1020)
	at java.base/java.util.concurrent.ForkJoinPool.scan(ForkJoinPool.java:1656)
	at java.base/java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1594)
	at java.base/java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:183)
Caused by: com.oracle.graal.pointsto.constraints.UnsupportedFeatureException: Unsupported features in 4 methods
Detailed message:
Error: com.oracle.graal.pointsto.constraints.UnsupportedFeatureException: Invoke with MethodHandle argument could not be reduced to at most a single call or single field access. The method handle must be a compile time constant, e.g., be loaded from a `static final` field. Method that contains the method handle invocation: java.lang.invoke.MethodHandle.invokeBasic(Object)
To enable method handles that do not require LambdaForm interpretation (e.g. because of a call to MethodHandle.bindTo()) or to diagnose the issue, you can add the option --report-unsupported-elements-at-runtime. The error is then reported at run time when the invoke is executed and the method handle has to be interpreted.
Trace: 
	at parsing java.lang.invoke.LambdaForm$MH/0x00000007c24ac040.invoke_MT(LambdaForm$MH)
Call path from entry point to java.lang.invoke.LambdaForm$MH/1574591223.invoke_MT(Object, Object, Object): 
	at java.lang.invoke.LambdaForm$MH/0x00000007c24ac040.invoke_MT(LambdaForm$MH)
	at org.hibernate.search.util.common.reflect.impl.MethodHandleValueReadHandle.get(MethodHandleValueReadHandle.java:39)
	at org.hibernate.search.mapper.pojo.bridge.runtime.impl.PropertyIdentifierMapping.getIdentifier(PropertyIdentifierMapping.java:60)
	at org.hibernate.search.mapper.pojo.work.impl.PojoTypeIndexer.add(PojoTypeIndexer.java:37)
	at org.hibernate.search.mapper.pojo.work.impl.PojoIndexerImpl.add(PojoIndexerImpl.java:42)
	at org.hibernate.search.mapper.orm.massindexing.impl.IdentifierConsumerDocumentProducer.index(IdentifierConsumerDocumentProducer.java:243)
	at org.hibernate.search.mapper.orm.massindexing.impl.IdentifierConsumerDocumentProducer.indexList(IdentifierConsumerDocumentProducer.java:201)
	at org.hibernate.search.mapper.orm.massindexing.impl.IdentifierConsumerDocumentProducer.loadAndIndexList(IdentifierConsumerDocumentProducer.java:154)
	at org.hibernate.search.mapper.orm.massindexing.impl.IdentifierConsumerDocumentProducer.loadAndIndexAllFromQueue(IdentifierConsumerDocumentProducer.java:123)
	at org.hibernate.search.mapper.orm.massindexing.impl.IdentifierConsumerDocumentProducer.run(IdentifierConsumerDocumentProducer.java:103)
	at java.lang.Thread.run(Thread.java:829)
	at com.oracle.svm.core.thread.JavaThreads.threadStartRoutine(JavaThreads.java:519)
	at com.oracle.svm.core.posix.thread.PosixJavaThreads.pthreadStartRoutine(PosixJavaThreads.java:192)
	at com.oracle.svm.core.code.IsolateEnterStub.PosixJavaThreads_pthreadStartRoutine_e1f4a8c0039f8337338252cd8734f63a79b5e3df(generated:0)
Error: com.oracle.svm.hosted.substitute.DeletedElementException: Unsupported type java.lang.invoke.MemberName is reachable: All methods from java.lang.invoke should have been replaced during image building.
To diagnose the issue, you can add the option --report-unsupported-elements-at-runtime. The unsupported element is then reported at run time when it is accessed the first time.
Trace: 
	at parsing java.lang.invoke.LambdaForm$NamedFunction.equals(LambdaForm.java:1177)
Call path from entry point to java.lang.invoke.LambdaForm$NamedFunction.equals(Object): 
	at java.lang.invoke.LambdaForm$NamedFunction.equals(LambdaForm.java:1173)
	at java.util.HashMap.getNode(HashMap.java:574)
	at java.util.HashMap.get(HashMap.java:552)
	at com.oracle.svm.core.jdk.Target_java_nio_charset_Charset.lookup(CharsetSubstitutions.java:74)
	at java.nio.charset.Charset.isSupported(Charset.java:503)
	at com.oracle.svm.jni.JNIJavaCallWrappers.jniInvoke_VA_LIST:Ljava_nio_charset_Charset_2_0002eisSupported_00028Ljava_lang_String_2_00029Z(generated:0)
Error: com.oracle.svm.hosted.substitute.DeletedElementException: Unsupported type java.lang.invoke.MemberName is reachable: All methods from java.lang.invoke should have been replaced during image building.
To diagnose the issue, you can add the option --report-unsupported-elements-at-runtime. The unsupported element is then reported at run time when it is accessed the first time.
Trace: 
	at parsing java.lang.invoke.LambdaForm$NamedFunction.hashCode(LambdaForm.java:1182)
Call path from entry point to java.lang.invoke.LambdaForm$NamedFunction.hashCode(): 
	at java.lang.invoke.LambdaForm$NamedFunction.hashCode(LambdaForm.java:1182)
	at java.util.HashMap.hash(HashMap.java:339)
	at java.util.HashMap.get(HashMap.java:552)
	at com.oracle.svm.core.jdk.Target_java_nio_charset_Charset.lookup(CharsetSubstitutions.java:74)
	at java.nio.charset.Charset.isSupported(Charset.java:503)
	at com.oracle.svm.jni.JNIJavaCallWrappers.jniInvoke_VA_LIST:Ljava_nio_charset_Charset_2_0002eisSupported_00028Ljava_lang_String_2_00029Z(generated:0)
Error: com.oracle.svm.hosted.substitute.DeletedElementException: Unsupported type java.lang.invoke.MemberName is reachable: All methods from java.lang.invoke should have been replaced during image building.
To diagnose the issue, you can add the option --report-unsupported-elements-at-runtime. The unsupported element is then reported at run time when it is accessed the first time.
Trace: 
	at parsing java.lang.invoke.LambdaForm$NamedFunction.toString(LambdaForm.java:1279)
Call path from entry point to java.lang.invoke.LambdaForm$NamedFunction.toString(): 
	at java.lang.invoke.LambdaForm$NamedFunction.toString(LambdaForm.java:1279)
	at java.lang.String.valueOf(String.java:2951)
	at java.io.PrintStream.print(PrintStream.java:745)
	at java.io.PrintStream.println(PrintStream.java:882)
	at com.oracle.svm.jni.functions.JNIFunctions.ExceptionDescribe(JNIFunctions.java:757)
	at com.oracle.svm.core.code.IsolateEnterStub.JNIFunctions_ExceptionDescribe_b5412f7570bccae90b000bc37855f00408b2ad73(generated:0)

	at com.oracle.graal.pointsto.constraints.UnsupportedFeatures.report(UnsupportedFeatures.java:129)
	at com.oracle.svm.hosted.NativeImageGenerator.runPointsToAnalysis(NativeImageGenerator.java:763)
	... 8 more
Error: Image build request failed with exit status 1
</pre>
</details>


## To Reproduce

```
git clone --depth 1 https://github.com/graalvm/mx.git
git clone --depth 1 --branch release/graal-vm/20.3 https://github.com/oracle/graal.git
cd graal
../mx/mx --primary-suite-path substratevm  --native=native-image --components='Native Image' build
export GRAALVM_HOME=$(pwd)/sdk/latest_graalvm_home/
cd path/to/quarkus
./mvnw clean verify -Dnative -Dnative.surefire.skip -pl integration-tests/hibernate-search-orm-elasticsearch
```

## Environment (please complete the following information):

```
┌────────────────────────────┬─────────────────────────────────────────────────┐
│uname -a                    │Linux         5.11.11-200.fc33.x86_64 #1 SMP Tue │
│                            │Mar 30 16:53:32 UTC 2021 x86_64 x86_64 x86_64    │
│                            │GNU/Linux                                        │
├────────────────────────────┼─────────────────────────────────────────────────┤
│java -version               │openjdk version "11.0.10" 2021-01-19             │
│                            │OpenJDK Runtime Environment 18.9 (build          │
│                            │11.0.10+9)                                       │
│                            │OpenJDK 64-Bit Server VM 18.9 (build 11.0.10+9,  │
│                            │mixed mode, sharing)                             │
├────────────────────────────┼─────────────────────────────────────────────────┤
│graalvm java -version       │openjdk version "11.0.11" 2021-04-20             │
│                            │OpenJDK Runtime Environment GraalVM 20.3.2-dev   │
│                            │(build 11.0.11+5-jvmci-21.1-b02)                 │
│                            │OpenJDK 64-Bit Server VM GraalVM 20.3.2-dev      │
│                            │(build 11.0.11+5-jvmci-21.1-b02, mixed mode,     │
│                            │sharing)                                         │
├────────────────────────────┼─────────────────────────────────────────────────┤
│mvnw --version              │Apache Maven 3.6.3                               │
│                            │(cecedd343002696d0abb50b32b541b8a6ba2883f)       │
│                            │Maven home:                                      │
│                            │/home/zakkak/.m2/wrapper/dists/apache-maven-3.6.3│
│                            │-bin/1iopthnavndlasol9gbrbg6bf2/apache-maven-3.6.│
│                            │3                                                │
│                            │Java version: 11.0.10, vendor: Red Hat, Inc.,    │
│                            │runtime:                                         │
│                            │/usr/lib/jvm/java-11-openjdk-11.0.10.0.9-0.fc33.x│
│                            │86_64                                            │
│                            │Default locale: en_IE, platform encoding: UTF-8  │
│                            │OS name: "linux", version:                       │
│                            │"5.11.11-200.fc33.x86_64", arch: "amd64", family:│
│                            │"unix"                                           │
└────────────────────────────┴─────────────────────────────────────────────────┘

--- a/extensions/hibernate-search-orm-elasticsearch/runtime/src/main/java/io/quarkus/hibernate/search/orm/elasticsearch/runtime/HibernateSearchElasticsearchRecorder.java
+++ b/extensions/hibernate-search-orm-elasticsearch/runtime/src/main/java/io/quarkus/hibernate/search/orm/elasticsearch/runtime/HibernateSearchElasticsearchRecorder.java
@@ -139,8 +139,8 @@ public class HibernateSearchElasticsearchRecorder {
         public void onMetadataInitialized(Metadata metadata, BootstrapContext bootstrapContext,
                 BiConsumer<String, Object> propertyCollector) {
             Version graalVMVersion = Version.getCurrent();
-            boolean isGraalVM20OrBelow = !graalVMVersion.isSnapshot() // isSnapshot() will be true on OpenJDK
-                    && graalVMVersion.compareTo(GRAAL_VM_VERSION_21) < 0;
+            final boolean isAOT = Boolean.getBoolean("com.oracle.graalvm.isaot");
+            boolean isGraalVM20OrBelow = isAOT && graalVMVersion.compareTo(GRAAL_VM_VERSION_21) < 0;
             HibernateOrmIntegrationBooter booter = HibernateOrmIntegrationBooter.builder(metadata, bootstrapContext)
                     .valueReadHandleFactory(
                             // GraalVM 20 or below doesn't support method handles

